### PR TITLE
added: filter functionality

### PR DESCRIPTION
**Pull request by rkapoor10**
added filter functionality

- folder structure
  
folder-structure

- Filters
   - Range Filter
   - Category Filter
   - Ratings 
   - Price Sort
   - Discount
   
   
filtersss

### CODE REVIEW

File: src/context/filterContext/filterReducer.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import React from "react";
+
```
Comment (JantuDeb) [inline src/context/filterContext/filterReducer.jsx]: You can create a variable 
```
export const initialState = {
    highToLow: false,
    lowToHigh: false,
    discount: {
      "10_OR_BELOW": false,
      "10_OR_ABOVE": false,
      "30_OR_ABOVE": false,
    },
    category: { racquets: false, shoes: false, apparel: false, bags: false },
    ratings: 0,
    priceRange: 2000,
  }

```
Use it for reset filter also

File: src/context/filterContext/filterReducer.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          },
+          category: {
+            racquets: false,
+            shoes: false,
+            apparel: false,
+            bags: false,
+          },
+          ratings: 0,
+          priceRange: 2000,
+        };
```
Comment (JantuDeb) [inline src/context/filterContext/filterReducer.jsx]: ``` return  initialState```

File: src/utils/filtersFunctions.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  }
+  return data;
+};
+
+const categoryFilter = (state, data) => {
+
+  const categoriesArray = Object.keys(state.category).filter((value)=>state.category[value] === true)
+  if(categoriesArray.length===0){
+    return data;
+  }
```
Comment (JantuDeb) [inline src/utils/filtersFunctions.js]: if(categoriesArray.length===0) return data;

File: src/utils/filtersFunctions.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    (products) => Number(products.ratings.stars) >= Number(state.ratings)
+  );
+};
+
+const priceRangeFilter = (state, data) => {
+  return data.filter(
+    (product) => Number(product.price) <= Number(state.priceRange)
+  );
+};
+
```
Comment (JantuDeb) [inline src/utils/filtersFunctions.js]:
```
const ratingsFilter = (state, data) =>  data.filter(
    (products) => Number(products.ratings.stars) >= Number(state.ratings) );


const priceRangeFilter = (state, data) =>  data.filter(
    (product) => Number(product.price) <= Number(state.priceRange) );


```

**Review by JantuDeb — APPROVED** [review]
Added some comments, you can check that.

File: src/utils/filtersFunctions.js (hunk 4 above)
Comment (rkapoor10) [reply src/utils/filtersFunctions.js]: done

File: src/utils/filtersFunctions.js (hunk 3 above)
Comment (rkapoor10) [reply src/utils/filtersFunctions.js]: done

File: src/context/filterContext/filterReducer.jsx (hunk 1 above)
Comment (rkapoor10) [reply src/context/filterContext/filterReducer.jsx]: done

File: src/context/filterContext/filterReducer.jsx (hunk 2 above)
Comment (rkapoor10) [reply src/context/filterContext/filterReducer.jsx]: done

File: src/components/filters/Filter.jsx [hunk 5]
```diff
@@ -1,93 +1,166 @@
...
+          >
+            Clear All
+          </button>
         </div>
+        <div className="seperator padtb-1">
+          <h4 className="mb-1 darksecondary">PRICE</h4>
+          <div style={{ width: "200px" }}>
+            <input
+              type="range"
+              style={{ width: "200px" }}
```
Comment (ShreyashMahajan) [inline src/components/filters/Filter.jsx:25]: avoid using inline css

**Review by ShreyashMahajan — APPROVED** [review]
looks good to me

File: src/components/filters/Filter.jsx (hunk 5 above)
Comment (rkapoor10) [reply src/components/filters/Filter.jsx:25]: okay

**Pull request merged**